File: Easy/1512. Number of Good Pairs/1512. Number of Good Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int[] countArray = new int[101]; // Assuming nums[i] <= 100
        int goodPairs = 0;

        for (int num : nums) {
            goodPairs += countArray[num];
            countArray[num]++;
        }

        return goodPairs;
    }
}

File: Easy/1512. Number of Good Pairs/1512. Number of Good Pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count_array = [0] * 101  # Assuming nums[i] <= 100
        good_pairs = 0

        for num in nums:
            good_pairs += count_array[num]
            count_array[num] += 1

        return good_pairs
